// lusid-client/models/addTradePropertyDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a AddTradePropertyDto.
 */
class AddTradePropertyDto {
  /**
   * Create a AddTradePropertyDto.
   * @member {string} [href]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of AddTradePropertyDto
   *
   * @returns {object} metadata of AddTradePropertyDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'AddTradePropertyDto',
      type: {
        name: 'Composite',
        className: 'AddTradePropertyDto',
        modelProperties: {
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = AddTradePropertyDto;
