// lusid-client/models/aggregateSpec.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a AggregateSpec.
 */
class AggregateSpec {
  /**
   * Create a AggregateSpec.
   * @member {string} [key]
   * @member {string} [op] Possible values include: 'Sum', 'Proportion'
   */
  constructor() {
  }

  /**
   * Defines the metadata of AggregateSpec
   *
   * @returns {object} metadata of AggregateSpec
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'AggregateSpec',
      type: {
        name: 'Composite',
        className: 'AggregateSpec',
        modelProperties: {
          key: {
            required: false,
            serializedName: 'key',
            type: {
              name: 'String'
            }
          },
          op: {
            required: false,
            serializedName: 'op',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = AggregateSpec;
