// lusid-client/models/aggregationResponseNodeIDataRecord.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a AggregationResponseNodeIDataRecord.
 */
class AggregationResponseNodeIDataRecord {
  /**
   * Create a AggregationResponseNodeIDataRecord.
   * @member {string} [description]
   * @member {number} [idx]
   * @member {object} [id]
   * @member {string} [id.id]
   * @member {string} [id.scope]
   * @member {string} [id.type] Possible values include: 'Group', 'Portfolio',
   * 'ReferencePortfolio'
   * @member {object} [properties]
   * @member {number} [properties.fieldCount]
   * @member {array} [children]
   */
  constructor() {
  }

  /**
   * Defines the metadata of AggregationResponseNodeIDataRecord
   *
   * @returns {object} metadata of AggregationResponseNodeIDataRecord
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'AggregationResponseNode_IDataRecord_',
      type: {
        name: 'Composite',
        className: 'AggregationResponseNodeIDataRecord',
        modelProperties: {
          description: {
            required: false,
            serializedName: 'description',
            type: {
              name: 'String'
            }
          },
          idx: {
            required: false,
            serializedName: 'idx',
            type: {
              name: 'Number'
            }
          },
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'Composite',
              className: 'QualifiedId'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Composite',
              className: 'IDataRecord'
            }
          },
          children: {
            required: false,
            serializedName: 'children',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'AggregationResponseNodeIDataRecordElementType',
                  type: {
                    name: 'Composite',
                    className: 'AggregationResponseNodeIDataRecord'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = AggregationResponseNodeIDataRecord;
