// lusid-client/models/clientSecurityDefinitionData.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a ClientSecurityDefinitionData.
 */
class ClientSecurityDefinitionData {
  /**
   * Create a ClientSecurityDefinitionData.
   * @member {string} [clientSecurityId]
   * @member {string} [name]
   * @member {array} [properties]
   * @member {array} [aliases]
   * @member {string} [lookThroughPortfolioId]
   * @member {string} [lookThroughPortfolioScope]
   * @member {object} [instrument] There could be multiple underlying
   * instrument definitions (same
   * instrument but different format), but for now store one.
   * @member {string} [instrument.content]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ClientSecurityDefinitionData
   *
   * @returns {object} metadata of ClientSecurityDefinitionData
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ClientSecurityDefinitionData',
      type: {
        name: 'Composite',
        className: 'ClientSecurityDefinitionData',
        modelProperties: {
          clientSecurityId: {
            required: false,
            serializedName: 'clientSecurityId',
            type: {
              name: 'String'
            }
          },
          name: {
            required: false,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          },
          aliases: {
            required: false,
            serializedName: 'aliases',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'KeyValuePairCodeTypeStringElementType',
                  type: {
                    name: 'Composite',
                    className: 'KeyValuePairCodeTypeString'
                  }
              }
            }
          },
          lookThroughPortfolioId: {
            required: false,
            serializedName: 'lookThroughPortfolioId',
            type: {
              name: 'String'
            }
          },
          lookThroughPortfolioScope: {
            required: false,
            serializedName: 'lookThroughPortfolioScope',
            type: {
              name: 'String'
            }
          },
          instrument: {
            required: false,
            serializedName: 'instrument',
            type: {
              name: 'Composite',
              className: 'InstrumentDefinition'
            }
          }
        }
      }
    };
  }
}

module.exports = ClientSecurityDefinitionData;
