// lusid-client/models/createPortfolioRequest.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a CreatePortfolioRequest.
 */
class CreatePortfolioRequest {
  /**
   * Create a CreatePortfolioRequest.
   * @member {object} [root]
   * @member {string} [root.id] Portfolio id, must be unique across all
   * portfolios
   * @member {string} [root.name] Portfolio name, must be unique across all
   * portfolios
   * @member {date} [root.created] The original creation date, not the
   * effective date
   * of subsequent modification.
   * @member {string} [root.referencePortfolioId] If creating a derived
   * portfolio, this is the portfolio id of the reference portfolio
   * @member {string} [root.referenceScope] If creating a derived portfolio,
   * this is the scope of the reference portfolio
   * @member {string} [root.type] Possible values include: 'Portfolio',
   * 'ReferencePortfolio'
   * @member {object} [details]
   * @member {string} [details.sourcePortfolioId] The portfolio id that these
   * portfolio details belong to, used when inheriting
   * from a reference portfolio
   * @member {string} [details.sourceScope] The scope that these portfolio
   * details belong to
   * @member {string} [details.baseCurrency]
   */
  constructor() {
  }

  /**
   * Defines the metadata of CreatePortfolioRequest
   *
   * @returns {object} metadata of CreatePortfolioRequest
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'CreatePortfolioRequest',
      type: {
        name: 'Composite',
        className: 'CreatePortfolioRequest',
        modelProperties: {
          root: {
            required: false,
            serializedName: 'root',
            type: {
              name: 'Composite',
              className: 'Portfolio'
            }
          },
          details: {
            required: false,
            serializedName: 'details',
            type: {
              name: 'Composite',
              className: 'PortfolioDetails'
            }
          }
        }
      }
    };
  }
}

module.exports = CreatePortfolioRequest;
