// lusid-client/models/createDerivedPortfolioRequest.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a CreateDerivedPortfolioRequest.
 */
class CreateDerivedPortfolioRequest {
  /**
   * Create a CreateDerivedPortfolioRequest.
   * @member {string} [id]
   * @member {object} [basePortfolio]
   * @member {string} [basePortfolio.scope]
   * @member {string} [basePortfolio.id]
   * @member {string} name Portfolio name (aka description
   * @member {date} [created] The original creation date, not the effective
   * date
   * of subsequent modification.
   */
  constructor() {
  }

  /**
   * Defines the metadata of CreateDerivedPortfolioRequest
   *
   * @returns {object} metadata of CreateDerivedPortfolioRequest
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'CreateDerivedPortfolioRequest',
      type: {
        name: 'Composite',
        className: 'CreateDerivedPortfolioRequest',
        modelProperties: {
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'String'
            }
          },
          basePortfolio: {
            required: false,
            serializedName: 'basePortfolio',
            type: {
              name: 'Composite',
              className: 'DtoId'
            }
          },
          name: {
            required: true,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          created: {
            required: false,
            serializedName: 'created',
            type: {
              name: 'DateTime'
            }
          }
        }
      }
    };
  }
}

module.exports = CreateDerivedPortfolioRequest;
