// lusid-client/models/dataScope.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Everything in the system is scoped, this class encapsulates this concept
 *
 */
class DataScope {
  /**
   * Create a DataScope.
   * @member {object} [client]
   * @member {string} [client.name]
   * @member {string} [scope]
   */
  constructor() {
  }

  /**
   * Defines the metadata of DataScope
   *
   * @returns {object} metadata of DataScope
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'DataScope',
      type: {
        name: 'Composite',
        className: 'DataScope',
        modelProperties: {
          client: {
            required: false,
            readOnly: true,
            serializedName: 'client',
            type: {
              name: 'Composite',
              className: 'Client'
            }
          },
          scope: {
            required: false,
            readOnly: true,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = DataScope;
