// lusid-client/models/errorMessage.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a ErrorMessage.
 */
class ErrorMessage {
  /**
   * Create a ErrorMessage.
   * @member {number} [status] The status code that will be returned to the
   * client
   * @member {string} [code] The Finbourne specific error-code that
   * encapsulates the specific issue encountered. Possible values include:
   * 'PersonalisationNotFound', 'NonRecursivePersonalisation',
   * 'VersionNotFound', 'TradeNotFound', 'SecurityNotFound',
   * 'PropertyNotFound', 'PortfolioRecursionDepth',
   * 'PortfolioPropertiesNotFound', 'GroupNotFound', 'PortfolioNotFound',
   * 'EntitySchemaNotFound', 'PortfolioNameNotFound', 'PortfolioAlreadyExists',
   * 'OrphanedPortfolio', 'MultiplePropertyDefinitionNotFound',
   * 'InvalidProperty', 'InvalidBasePortfolioAction', 'InsufficientAuthClaims',
   * 'MissingBaseClaims', 'MissingIdentity', 'PropertyNotDefined',
   * 'CannotDeleteSystemProperty', 'CannotModifyImmutablePropertyField',
   * 'PropertyAlreadyExists', 'InvalidPropertyLifeTime',
   * 'RequiredFieldIsMissing', 'CannotModifyDefaultPropertyFormat',
   * 'GroupAlreadyExists', 'NoSuchPropertyDataFormat',
   * 'CannotAddMissingItemToGroup', 'CannotAddAGroupToItself',
   * 'ValidationError', 'LoopDetectedInGroupHierarchy',
   * 'AddingSubGroupCreatedALoop', 'SubGroupAlreadyExists',
   * 'PortfolioAlreadyExistsInGroup', 'MissingClassificationData',
   * 'PriceSourceNotFound', 'AnalyticStoreNotFound', 'InvalidDateFormat',
   * 'AnalyticStoreAlreadyExists', 'ClientSecurityAlreadyExists',
   * 'DuplicateParameterException', 'OneOrMoreFailures', 'OneOrMoreMissing',
   * 'ResultsNotFound', 'OrderFieldNotInResultSet', 'OperationFailedEnum',
   * 'ElasticSearchError', 'InvalidParameterValue', 'Unknown'
   * @member {string} [property] The part of the resource which provoked the
   * error (e.g. field)
   * @member {string} [message] The non-technical-user friendly message
   * describing the error and how it might be remedied.
   * @member {string} [developerMessage] A technical error message that
   * contains the details of the issue and how it might be fixed.
   * @member {string} [moreInfo] A link to the endpoint that can provide the
   * dev with more information about that class of error.
   */
  constructor() {
  }

  /**
   * Defines the metadata of ErrorMessage
   *
   * @returns {object} metadata of ErrorMessage
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ErrorMessage',
      type: {
        name: 'Composite',
        className: 'ErrorMessage',
        modelProperties: {
          status: {
            required: false,
            readOnly: true,
            serializedName: 'status',
            type: {
              name: 'Number'
            }
          },
          code: {
            required: false,
            readOnly: true,
            serializedName: 'code',
            type: {
              name: 'String'
            }
          },
          property: {
            required: false,
            readOnly: true,
            serializedName: 'property',
            type: {
              name: 'String'
            }
          },
          message: {
            required: false,
            readOnly: true,
            serializedName: 'message',
            type: {
              name: 'String'
            }
          },
          developerMessage: {
            required: false,
            readOnly: true,
            serializedName: 'developerMessage',
            type: {
              name: 'String'
            }
          },
          moreInfo: {
            required: false,
            readOnly: true,
            serializedName: 'moreInfo',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ErrorMessage;
