// lusid-client/models/getLoginResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a GetLoginResponse.
 */
class GetLoginResponse {
  /**
   * Create a GetLoginResponse.
   * @member {object} [error]
   * @member {number} [error.status] The status code that will be returned to
   * the client
   * @member {string} [error.code] The Finbourne specific error-code that
   * encapsulates the specific issue encountered. Possible values include:
   * 'PersonalisationNotFound', 'NonRecursivePersonalisation',
   * 'VersionNotFound', 'TradeNotFound', 'SecurityNotFound',
   * 'PropertyNotFound', 'PortfolioRecursionDepth',
   * 'PortfolioPropertiesNotFound', 'GroupNotFound', 'PortfolioNotFound',
   * 'EntitySchemaNotFound', 'PortfolioNameNotFound', 'PortfolioAlreadyExists',
   * 'OrphanedPortfolio', 'MultiplePropertyDefinitionNotFound',
   * 'InvalidProperty', 'InvalidBasePortfolioAction', 'InsufficientAuthClaims',
   * 'MissingBaseClaims', 'MissingIdentity', 'PropertyNotDefined',
   * 'CannotDeleteSystemProperty', 'CannotModifyImmutablePropertyField',
   * 'PropertyAlreadyExists', 'InvalidPropertyLifeTime',
   * 'RequiredFieldIsMissing', 'CannotModifyDefaultPropertyFormat',
   * 'GroupAlreadyExists', 'NoSuchPropertyDataFormat',
   * 'CannotAddMissingItemToGroup', 'CannotAddAGroupToItself',
   * 'ValidationError', 'LoopDetectedInGroupHierarchy',
   * 'AddingSubGroupCreatedALoop', 'SubGroupAlreadyExists',
   * 'PortfolioAlreadyExistsInGroup', 'MissingClassificationData',
   * 'PriceSourceNotFound', 'AnalyticStoreNotFound', 'InvalidDateFormat',
   * 'AnalyticStoreAlreadyExists', 'ClientSecurityAlreadyExists',
   * 'DuplicateParameterException', 'OneOrMoreFailures', 'OneOrMoreMissing',
   * 'ResultsNotFound', 'OrderFieldNotInResultSet', 'OperationFailedEnum',
   * 'ElasticSearchError', 'InvalidParameterValue', 'Unknown'
   * @member {string} [error.property] The part of the resource which provoked
   * the error (e.g. field)
   * @member {string} [error.message] The non-technical-user friendly message
   * describing the error and how it might be remedied.
   * @member {string} [error.developerMessage] A technical error message that
   * contains the details of the issue and how it might be fixed.
   * @member {string} [error.moreInfo] A link to the endpoint that can provide
   * the dev with more information about that class of error.
   * @member {object} [meta]
   * @member {number} [meta.duration]
   * @member {boolean} [meta.success]
   * @member {string} [meta.requestId]
   * @member {string} [self]
   * @member {string} [schema]
   * @member {string} [name]
   * @member {string} [firstName]
   * @member {string} [defaultScope]
   * @member {string} [sessionContextId]
   * @member {string} [logoUri]
   * @member {string} [valueCurrencyCode]
   * @member {string} [valueCurrencySymbol]
   */
  constructor() {
  }

  /**
   * Defines the metadata of GetLoginResponse
   *
   * @returns {object} metadata of GetLoginResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'GetLoginResponse',
      type: {
        name: 'Composite',
        className: 'GetLoginResponse',
        modelProperties: {
          error: {
            required: false,
            serializedName: 'error',
            type: {
              name: 'Composite',
              className: 'ErrorMessage'
            }
          },
          meta: {
            required: false,
            serializedName: 'meta',
            type: {
              name: 'Composite',
              className: 'ResponseMeta'
            }
          },
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          name: {
            required: false,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          firstName: {
            required: false,
            serializedName: 'firstName',
            type: {
              name: 'String'
            }
          },
          defaultScope: {
            required: false,
            serializedName: 'defaultScope',
            type: {
              name: 'String'
            }
          },
          sessionContextId: {
            required: false,
            serializedName: 'sessionContextId',
            type: {
              name: 'String'
            }
          },
          logoUri: {
            required: false,
            serializedName: 'logoUri',
            type: {
              name: 'String'
            }
          },
          valueCurrencyCode: {
            required: false,
            serializedName: 'valueCurrencyCode',
            type: {
              name: 'String'
            }
          },
          valueCurrencySymbol: {
            required: false,
            serializedName: 'valueCurrencySymbol',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = GetLoginResponse;
