// lusid-client/models/getSecurityRootResponseAliases.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a GetSecurityRootResponseAliases.
 */
class GetSecurityRootResponseAliases {
  /**
   * Create a GetSecurityRootResponseAliases.
   * @member {string} [undefined]
   * @member {string} [reutersAssetId]
   * @member {string} [cINS]
   * @member {string} [isin]
   * @member {string} [sedol]
   * @member {string} [cusip]
   * @member {string} [clientInternal]
   * @member {string} [figi]
   * @member {string} [wertpapier]
   */
  constructor() {
  }

  /**
   * Defines the metadata of GetSecurityRootResponseAliases
   *
   * @returns {object} metadata of GetSecurityRootResponseAliases
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'GetSecurityRootResponse_aliases',
      type: {
        name: 'Composite',
        className: 'GetSecurityRootResponseAliases',
        modelProperties: {
          undefined: {
            required: false,
            serializedName: 'Undefined',
            type: {
              name: 'String'
            }
          },
          reutersAssetId: {
            required: false,
            serializedName: 'ReutersAssetId',
            type: {
              name: 'String'
            }
          },
          cINS: {
            required: false,
            serializedName: 'CINS',
            type: {
              name: 'String'
            }
          },
          isin: {
            required: false,
            serializedName: 'Isin',
            type: {
              name: 'String'
            }
          },
          sedol: {
            required: false,
            serializedName: 'Sedol',
            type: {
              name: 'String'
            }
          },
          cusip: {
            required: false,
            serializedName: 'Cusip',
            type: {
              name: 'String'
            }
          },
          clientInternal: {
            required: false,
            serializedName: 'ClientInternal',
            type: {
              name: 'String'
            }
          },
          figi: {
            required: false,
            serializedName: 'Figi',
            type: {
              name: 'String'
            }
          },
          wertpapier: {
            required: false,
            serializedName: 'Wertpapier',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = GetSecurityRootResponseAliases;
