// lusid-client/models/holding.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a Holding.
 */
class Holding {
  /**
   * Create a Holding.
   * @member {string} securityUid Id of the underlying security
   * @member {array} [properties]
   * @member {number} units Total number of units in the holding
   * @member {number} cost Total cost of the holding
   */
  constructor() {
  }

  /**
   * Defines the metadata of Holding
   *
   * @returns {object} metadata of Holding
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Holding',
      type: {
        name: 'Composite',
        className: 'Holding',
        modelProperties: {
          securityUid: {
            required: true,
            serializedName: 'securityUid',
            type: {
              name: 'String'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          },
          units: {
            required: true,
            serializedName: 'units',
            type: {
              name: 'Number'
            }
          },
          cost: {
            required: true,
            serializedName: 'cost',
            type: {
              name: 'Number'
            }
          }
        }
      }
    };
  }
}

module.exports = Holding;
