// lusid-client/models/keyValuePairCodeTypeString.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a KeyValuePairCodeTypeString.
 */
class KeyValuePairCodeTypeString {
  /**
   * Create a KeyValuePairCodeTypeString.
   * @member {string} [key] Possible values include: 'Undefined',
   * 'ReutersAssetId', 'CINS', 'Isin', 'Sedol', 'Cusip', 'ClientInternal',
   * 'Figi', 'Wertpapier'
   * @member {string} [value]
   */
  constructor() {
  }

  /**
   * Defines the metadata of KeyValuePairCodeTypeString
   *
   * @returns {object} metadata of KeyValuePairCodeTypeString
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'KeyValuePair_CodeType_String_',
      type: {
        name: 'Composite',
        className: 'KeyValuePairCodeTypeString',
        modelProperties: {
          key: {
            required: false,
            readOnly: true,
            serializedName: 'key',
            type: {
              name: 'String'
            }
          },
          value: {
            required: false,
            readOnly: true,
            serializedName: 'value',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = KeyValuePairCodeTypeString;
