// lusid-client/models/keyValuePairStringDateTimeOffset.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a KeyValuePairStringDateTimeOffset.
 */
class KeyValuePairStringDateTimeOffset {
  /**
   * Create a KeyValuePairStringDateTimeOffset.
   * @member {string} [key]
   * @member {date} [value]
   */
  constructor() {
  }

  /**
   * Defines the metadata of KeyValuePairStringDateTimeOffset
   *
   * @returns {object} metadata of KeyValuePairStringDateTimeOffset
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'KeyValuePair_String_DateTimeOffset_',
      type: {
        name: 'Composite',
        className: 'KeyValuePairStringDateTimeOffset',
        modelProperties: {
          key: {
            required: false,
            readOnly: true,
            serializedName: 'key',
            type: {
              name: 'String'
            }
          },
          value: {
            required: false,
            readOnly: true,
            serializedName: 'value',
            type: {
              name: 'DateTime'
            }
          }
        }
      }
    };
  }
}

module.exports = KeyValuePairStringDateTimeOffset;
