// lusid-client/models/link.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a Link.
 */
class Link {
  /**
   * Create a Link.
   * @member {string} relation Possible values include: 'Root', 'Properties',
   * 'Trades', 'Details'
   * @member {string} href
   * @member {string} [description]
   * @member {string} method Possible values include: 'POST', 'GET', 'PATCH',
   * 'DELETE'
   */
  constructor() {
  }

  /**
   * Defines the metadata of Link
   *
   * @returns {object} metadata of Link
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Link',
      type: {
        name: 'Composite',
        className: 'Link',
        modelProperties: {
          relation: {
            required: true,
            serializedName: 'relation',
            type: {
              name: 'String'
            }
          },
          href: {
            required: true,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          description: {
            required: false,
            serializedName: 'description',
            type: {
              name: 'String'
            }
          },
          method: {
            required: true,
            serializedName: 'method',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = Link;
