// lusid-client/models/nestedDataAggregationResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a NestedDataAggregationResponse.
 */
class NestedDataAggregationResponse {
  /**
   * Create a NestedDataAggregationResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {object} [error]
   * @member {number} [error.status] The status code that will be returned to
   * the client
   * @member {string} [error.code] The Finbourne specific error-code that
   * encapsulates the specific issue encountered. Possible values include:
   * 'PersonalisationNotFound', 'NonRecursivePersonalisation',
   * 'VersionNotFound', 'TradeNotFound', 'SecurityNotFound',
   * 'PropertyNotFound', 'PortfolioRecursionDepth',
   * 'PortfolioPropertiesNotFound', 'GroupNotFound', 'PortfolioNotFound',
   * 'EntitySchemaNotFound', 'PortfolioNameNotFound', 'PortfolioAlreadyExists',
   * 'OrphanedPortfolio', 'MultiplePropertyDefinitionNotFound',
   * 'InvalidProperty', 'InvalidBasePortfolioAction', 'InsufficientAuthClaims',
   * 'MissingBaseClaims', 'MissingIdentity', 'PropertyNotDefined',
   * 'CannotDeleteSystemProperty', 'CannotModifyImmutablePropertyField',
   * 'PropertyAlreadyExists', 'InvalidPropertyLifeTime',
   * 'RequiredFieldIsMissing', 'CannotModifyDefaultPropertyFormat',
   * 'GroupAlreadyExists', 'NoSuchPropertyDataFormat',
   * 'CannotAddMissingItemToGroup', 'CannotAddAGroupToItself',
   * 'ValidationError', 'LoopDetectedInGroupHierarchy',
   * 'AddingSubGroupCreatedALoop', 'SubGroupAlreadyExists',
   * 'PortfolioAlreadyExistsInGroup', 'MissingClassificationData',
   * 'PriceSourceNotFound', 'AnalyticStoreNotFound', 'InvalidDateFormat',
   * 'AnalyticStoreAlreadyExists', 'ClientSecurityAlreadyExists',
   * 'DuplicateParameterException', 'OneOrMoreFailures', 'OneOrMoreMissing',
   * 'ResultsNotFound', 'OrderFieldNotInResultSet', 'OperationFailedEnum',
   * 'ElasticSearchError', 'InvalidParameterValue', 'Unknown'
   * @member {string} [error.property] The part of the resource which provoked
   * the error (e.g. field)
   * @member {string} [error.message] The non-technical-user friendly message
   * describing the error and how it might be remedied.
   * @member {string} [error.developerMessage] A technical error message that
   * contains the details of the issue and how it might be fixed.
   * @member {string} [error.moreInfo] A link to the endpoint that can provide
   * the dev with more information about that class of error.
   * @member {object} [meta]
   * @member {number} [meta.duration]
   * @member {boolean} [meta.success]
   * @member {string} [meta.requestId]
   * @member {object} [data]
   * @member {string} [data.description]
   * @member {number} [data.idx]
   * @member {object} [data.id]
   * @member {string} [data.id.id]
   * @member {string} [data.id.scope]
   * @member {string} [data.id.type] Possible values include: 'Group',
   * 'Portfolio', 'ReferencePortfolio'
   * @member {object} [data.properties]
   * @member {number} [data.properties.fieldCount]
   * @member {array} [data.children]
   * @member {string} [aggregationCurrency]
   * @member {object} [dataSchema]
   * @member {array} [dataSchema.nodeValueSchema]
   * @member {object} [dataSchema.propertySchema]
   */
  constructor() {
  }

  /**
   * Defines the metadata of NestedDataAggregationResponse
   *
   * @returns {object} metadata of NestedDataAggregationResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'NestedDataAggregationResponse',
      type: {
        name: 'Composite',
        className: 'NestedDataAggregationResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          error: {
            required: false,
            serializedName: 'error',
            type: {
              name: 'Composite',
              className: 'ErrorMessage'
            }
          },
          meta: {
            required: false,
            serializedName: 'meta',
            type: {
              name: 'Composite',
              className: 'ResponseMeta'
            }
          },
          data: {
            required: false,
            serializedName: 'data',
            type: {
              name: 'Composite',
              className: 'AggregationResponseNodeIDataRecord'
            }
          },
          aggregationCurrency: {
            required: false,
            serializedName: 'aggregationCurrency',
            type: {
              name: 'String'
            }
          },
          dataSchema: {
            required: false,
            serializedName: 'dataSchema',
            type: {
              name: 'Composite',
              className: 'ResultDataSchema'
            }
          }
        }
      }
    };
  }
}

module.exports = NestedDataAggregationResponse;
