// lusid-client/models/personalisation.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a Personalisation.
 */
class Personalisation {
  /**
   * Create a Personalisation.
   * @member {number} [id]
   * @member {string} [scope] Possible values include: 'User', 'Group',
   * 'Default', 'All'
   * @member {string} [scopeValue]
   * @member {string} [settingKey]
   * @member {string} [settingValue]
   * @member {string} [dataType]
   */
  constructor() {
  }

  /**
   * Defines the metadata of Personalisation
   *
   * @returns {object} metadata of Personalisation
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Personalisation',
      type: {
        name: 'Composite',
        className: 'Personalisation',
        modelProperties: {
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'Number'
            }
          },
          scope: {
            required: false,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          },
          scopeValue: {
            required: false,
            serializedName: 'scopeValue',
            type: {
              name: 'String'
            }
          },
          settingKey: {
            required: false,
            serializedName: 'settingKey',
            type: {
              name: 'String'
            }
          },
          settingValue: {
            required: false,
            serializedName: 'settingValue',
            type: {
              name: 'String'
            }
          },
          dataType: {
            required: false,
            serializedName: 'dataType',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = Personalisation;
