// lusid-client/models/portfolioRootResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a PortfolioRootResponse.
 */
class PortfolioRootResponse {
  /**
   * Create a PortfolioRootResponse.
   * @member {string} [self] Link to retrieve the current entity
   * @member {string} [schema] Link to retrieve the schema for the current
   * entity
   * @member {string} [id] Id of the portfolio
   * @member {string} [name] Name of the portfolio
   * @member {date} [created] Portfolio creation time in UTC
   * @member {string} [referencePortfolioId]
   * @member {string} [referenceScope]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioRootResponse
   *
   * @returns {object} metadata of PortfolioRootResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioRootResponse',
      type: {
        name: 'Composite',
        className: 'PortfolioRootResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'String'
            }
          },
          name: {
            required: false,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          created: {
            required: false,
            serializedName: 'created',
            type: {
              name: 'DateTime'
            }
          },
          referencePortfolioId: {
            required: false,
            serializedName: 'referencePortfolioId',
            type: {
              name: 'String'
            }
          },
          referenceScope: {
            required: false,
            serializedName: 'referenceScope',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioRootResponse;
