// lusid-client/models/portfolioGroupState.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * DTO class used when creating a new portfolio group
 *
 */
class PortfolioGroupState {
  /**
   * Create a PortfolioGroupState.
   * @member {string} [name]
   * @member {string} [description]
   * @member {array} [values]
   * @member {array} [subGroups]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioGroupState
   *
   * @returns {object} metadata of PortfolioGroupState
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioGroupState',
      type: {
        name: 'Composite',
        className: 'PortfolioGroupState',
        modelProperties: {
          name: {
            required: false,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          description: {
            required: false,
            serializedName: 'description',
            type: {
              name: 'String'
            }
          },
          values: {
            required: false,
            serializedName: 'values',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'ScopedIdentifierElementType',
                  type: {
                    name: 'Composite',
                    className: 'ScopedIdentifier'
                  }
              }
            }
          },
          subGroups: {
            required: false,
            serializedName: 'subGroups',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'ScopedIdentifierElementType',
                  type: {
                    name: 'Composite',
                    className: 'ScopedIdentifier'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioGroupState;
