// lusid-client/models/portfolioDetailsResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a PortfolioDetailsResponse.
 */
class PortfolioDetailsResponse {
  /**
   * Create a PortfolioDetailsResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {string} [baseCurrency]
   * @member {string} [sourcePortfolioId]
   * @member {string} [sourceScope]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioDetailsResponse
   *
   * @returns {object} metadata of PortfolioDetailsResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioDetailsResponse',
      type: {
        name: 'Composite',
        className: 'PortfolioDetailsResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          baseCurrency: {
            required: false,
            serializedName: 'baseCurrency',
            type: {
              name: 'String'
            }
          },
          sourcePortfolioId: {
            required: false,
            serializedName: 'sourcePortfolioId',
            type: {
              name: 'String'
            }
          },
          sourceScope: {
            required: false,
            serializedName: 'sourceScope',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioDetailsResponse;
