// lusid-client/models/portfolioPropertiesResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a PortfolioPropertiesResponse.
 */
class PortfolioPropertiesResponse {
  /**
   * Create a PortfolioPropertiesResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {array} [propertySchema]
   * @member {string} [sourcePortfolioId]
   * @member {string} [sourceScope]
   * @member {array} [properties]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioPropertiesResponse
   *
   * @returns {object} metadata of PortfolioPropertiesResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioPropertiesResponse',
      type: {
        name: 'Composite',
        className: 'PortfolioPropertiesResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          propertySchema: {
            required: false,
            serializedName: 'propertySchema',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'KeyValuePairPropertyKeyFieldSchemaElementType',
                  type: {
                    name: 'Composite',
                    className: 'KeyValuePairPropertyKeyFieldSchema'
                  }
              }
            }
          },
          sourcePortfolioId: {
            required: false,
            serializedName: 'sourcePortfolioId',
            type: {
              name: 'String'
            }
          },
          sourceScope: {
            required: false,
            serializedName: 'sourceScope',
            type: {
              name: 'String'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioPropertiesResponse;
