// lusid-client/models/portfolioPropertiesDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a PortfolioPropertiesDto.
 */
class PortfolioPropertiesDto {
  /**
   * Create a PortfolioPropertiesDto.
   * @member {string} [href]
   * @member {object} [sourcePortfolioId]
   * @member {string} [sourcePortfolioId.scope]
   * @member {string} [sourcePortfolioId.id]
   * @member {array} [properties]
   * @member {object} [version] The version of the portfolio
   * @member {date} [version.validFromDate] The time at which this version
   * became valid.
   * @member {object} [version.asAtDate] The time at which this version was
   * durably persisted.
   * @member {string} [version.updatedBy] The name of the person or system
   * which created this version.
   * @member {string} [version.href]
   * @member {array} [version._links]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioPropertiesDto
   *
   * @returns {object} metadata of PortfolioPropertiesDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioPropertiesDto',
      type: {
        name: 'Composite',
        className: 'PortfolioPropertiesDto',
        modelProperties: {
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          sourcePortfolioId: {
            required: false,
            serializedName: 'sourcePortfolioId',
            type: {
              name: 'Composite',
              className: 'DtoId'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          },
          version: {
            required: false,
            serializedName: 'version',
            type: {
              name: 'Composite',
              className: 'VersionDto'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioPropertiesDto;
