// lusid-client/models/portfolio.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * This represents a portfolio owned by a client
 *
 */
class Portfolio {
  /**
   * Create a Portfolio.
   * @member {string} id Portfolio id, must be unique across all portfolios
   * @member {string} name Portfolio name, must be unique across all portfolios
   * @member {date} [created] The original creation date, not the effective
   * date
   * of subsequent modification.
   * @member {string} [referencePortfolioId] If creating a derived portfolio,
   * this is the portfolio id of the reference portfolio
   * @member {string} [referenceScope] If creating a derived portfolio, this is
   * the scope of the reference portfolio
   * @member {string} [type] Possible values include: 'Portfolio',
   * 'ReferencePortfolio'
   */
  constructor() {
  }

  /**
   * Defines the metadata of Portfolio
   *
   * @returns {object} metadata of Portfolio
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Portfolio',
      type: {
        name: 'Composite',
        className: 'Portfolio',
        modelProperties: {
          id: {
            required: true,
            serializedName: 'id',
            type: {
              name: 'String'
            }
          },
          name: {
            required: true,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          created: {
            required: false,
            serializedName: 'created',
            type: {
              name: 'DateTime'
            }
          },
          referencePortfolioId: {
            required: false,
            serializedName: 'referencePortfolioId',
            type: {
              name: 'String'
            }
          },
          referenceScope: {
            required: false,
            serializedName: 'referenceScope',
            type: {
              name: 'String'
            }
          },
          type: {
            required: false,
            serializedName: 'type',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = Portfolio;
