// lusid-client/models/portfolioDetails.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * This represents the details of a portfolio
 *
 */
class PortfolioDetails {
  /**
   * Create a PortfolioDetails.
   * @member {string} [sourcePortfolioId] The portfolio id that these portfolio
   * details belong to, used when inheriting
   * from a reference portfolio
   * @member {string} [sourceScope] The scope that these portfolio details
   * belong to
   * @member {string} baseCurrency
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioDetails
   *
   * @returns {object} metadata of PortfolioDetails
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioDetails',
      type: {
        name: 'Composite',
        className: 'PortfolioDetails',
        modelProperties: {
          sourcePortfolioId: {
            required: false,
            serializedName: 'sourcePortfolioId',
            type: {
              name: 'String'
            }
          },
          sourceScope: {
            required: false,
            serializedName: 'sourceScope',
            type: {
              name: 'String'
            }
          },
          baseCurrency: {
            required: true,
            serializedName: 'baseCurrency',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioDetails;
