// lusid-client/models/portfolioId.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * TODO: Should be moved to Movements project
 *
 */
class PortfolioId {
  /**
   * Create a PortfolioId.
   * @member {object} [scope]
   * @member {object} [scope.client]
   * @member {string} [scope.client.name]
   * @member {string} [scope.scope]
   * @member {string} [identifier]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioId
   *
   * @returns {object} metadata of PortfolioId
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioId',
      type: {
        name: 'Composite',
        className: 'PortfolioId',
        modelProperties: {
          scope: {
            required: false,
            readOnly: true,
            serializedName: 'scope',
            type: {
              name: 'Composite',
              className: 'DataScope'
            }
          },
          identifier: {
            required: false,
            readOnly: true,
            serializedName: 'identifier',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioId;
