// lusid-client/models/portfolioHoldingsDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a PortfolioHoldingsDto.
 */
class PortfolioHoldingsDto {
  /**
   * Create a PortfolioHoldingsDto.
   * @member {string} [href]
   * @member {array} [items]
   * @member {object} [version]
   * @member {date} [version.validFromDate] The time at which this version
   * became valid.
   * @member {object} [version.asAtDate] The time at which this version was
   * durably persisted.
   * @member {string} [version.updatedBy] The name of the person or system
   * which created this version.
   * @member {string} [version.href]
   * @member {array} [version._links]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PortfolioHoldingsDto
   *
   * @returns {object} metadata of PortfolioHoldingsDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PortfolioHoldingsDto',
      type: {
        name: 'Composite',
        className: 'PortfolioHoldingsDto',
        modelProperties: {
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          items: {
            required: false,
            serializedName: 'items',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'HoldingElementType',
                  type: {
                    name: 'Composite',
                    className: 'Holding'
                  }
              }
            }
          },
          version: {
            required: false,
            serializedName: 'version',
            type: {
              name: 'Composite',
              className: 'VersionDto'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = PortfolioHoldingsDto;
