// lusid-client/models/propertyDataFormatResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a PropertyDataFormatResponse.
 */
class PropertyDataFormatResponse {
  /**
   * Create a PropertyDataFormatResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {object} [item]
   * @member {string} [item.formatType] Possible values include: 'Basic',
   * 'Limited', 'Currency'
   * @member {string} [item.scope]
   * @member {string} [item.name]
   * @member {number} [item.order]
   * @member {string} [item.displayName]
   * @member {string} [item.valueType] Possible values include: 'String',
   * 'Int', 'Decimal', 'DateTime', 'Boolean', 'Map', 'PropertyArray',
   * 'ArrayOfKeys', 'Percentage', 'Currency', 'BenchmarkType', 'InternalId'
   */
  constructor() {
  }

  /**
   * Defines the metadata of PropertyDataFormatResponse
   *
   * @returns {object} metadata of PropertyDataFormatResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PropertyDataFormatResponse',
      type: {
        name: 'Composite',
        className: 'PropertyDataFormatResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          item: {
            required: false,
            serializedName: 'item',
            type: {
              name: 'Composite',
              className: 'PropertyDataFormat'
            }
          }
        }
      }
    };
  }
}

module.exports = PropertyDataFormatResponse;
