// lusid-client/models/propertyFilter.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a PropertyFilter.
 */
class PropertyFilter {
  /**
   * Create a PropertyFilter.
   * @member {string} [left]
   * @member {string} [operator] Possible values include: 'Equals',
   * 'NotEquals', 'GreaterThan', 'GreaterThanOrEqualTo', 'LessThan',
   * 'LessThanOrEqualTo', 'In'
   * @member {object} [right]
   * @member {string} [rightOperandType] Possible values include: 'Absolute',
   * 'Property'
   */
  constructor() {
  }

  /**
   * Defines the metadata of PropertyFilter
   *
   * @returns {object} metadata of PropertyFilter
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PropertyFilter',
      type: {
        name: 'Composite',
        className: 'PropertyFilter',
        modelProperties: {
          left: {
            required: false,
            serializedName: 'left',
            type: {
              name: 'String'
            }
          },
          operator: {
            required: false,
            serializedName: 'operator',
            type: {
              name: 'String'
            }
          },
          right: {
            required: false,
            serializedName: 'right',
            type: {
              name: 'Object'
            }
          },
          rightOperandType: {
            required: false,
            serializedName: 'rightOperandType',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PropertyFilter;
