// lusid-client/models/propertyDefinitionKeyResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a PropertyDefinitionKeyResponse.
 */
class PropertyDefinitionKeyResponse {
  /**
   * Create a PropertyDefinitionKeyResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {string} [scope]
   * @member {string} [name]
   * @member {string} [key]
   * @member {string} [domain] Possible values include: 'Trade', 'Portfolio',
   * 'Security', 'Holding', 'ReferenceHolding'
   */
  constructor() {
  }

  /**
   * Defines the metadata of PropertyDefinitionKeyResponse
   *
   * @returns {object} metadata of PropertyDefinitionKeyResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PropertyDefinitionKeyResponse',
      type: {
        name: 'Composite',
        className: 'PropertyDefinitionKeyResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          scope: {
            required: false,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          },
          name: {
            required: false,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          key: {
            required: false,
            serializedName: 'key',
            type: {
              name: 'String'
            }
          },
          domain: {
            required: false,
            serializedName: 'domain',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PropertyDefinitionKeyResponse;
