// lusid-client/models/property.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * This is intended to be the external facing property specification data type.
 *
 */
class Property {
  /**
   * Create a Property.
   * @member {string} key The property key made up of the PropertyDomain, scope
   * and name delimited with a '/'
   * e.g. trade/myscope/myproperty
   * @member {object} value
   * @member {date} [effectiveDate] Date for which the property is effective
   * from
   * @member {number} [version]
   */
  constructor() {
  }

  /**
   * Defines the metadata of Property
   *
   * @returns {object} metadata of Property
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Property',
      type: {
        name: 'Composite',
        className: 'Property',
        modelProperties: {
          key: {
            required: true,
            serializedName: 'key',
            type: {
              name: 'String'
            }
          },
          value: {
            required: true,
            serializedName: 'value',
            type: {
              name: 'Object'
            }
          },
          effectiveDate: {
            required: false,
            serializedName: 'effectiveDate',
            type: {
              name: 'DateTime'
            }
          },
          version: {
            required: false,
            serializedName: 'version',
            type: {
              name: 'Number'
            }
          }
        }
      }
    };
  }
}

module.exports = Property;
