// lusid-client/models/propertyScopeResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a PropertyScopeResponse.
 */
class PropertyScopeResponse {
  /**
   * Create a PropertyScopeResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {string} [scope]
   */
  constructor() {
  }

  /**
   * Defines the metadata of PropertyScopeResponse
   *
   * @returns {object} metadata of PropertyScopeResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'PropertyScopeResponse',
      type: {
        name: 'Composite',
        className: 'PropertyScopeResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          scope: {
            required: false,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = PropertyScopeResponse;
