// lusid-client/models/proxiedSearchResult.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a ProxiedSearchResult.
 */
class ProxiedSearchResult {
  /**
   * Create a ProxiedSearchResult.
   * @member {string} [body]
   * @member {number} [responseCode]
   * @member {string} [error]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ProxiedSearchResult
   *
   * @returns {object} metadata of ProxiedSearchResult
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ProxiedSearchResult',
      type: {
        name: 'Composite',
        className: 'ProxiedSearchResult',
        modelProperties: {
          body: {
            required: false,
            readOnly: true,
            serializedName: 'body',
            type: {
              name: 'String'
            }
          },
          responseCode: {
            required: false,
            readOnly: true,
            serializedName: 'responseCode',
            type: {
              name: 'Number'
            }
          },
          error: {
            required: false,
            readOnly: true,
            serializedName: 'error',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ProxiedSearchResult;
