// lusid-client/models/qualifiedId.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * This is an identifier that can be used to indicate not only an id, but its
 * scope, and the entity type to which it refers. This can be passed back to
 * the client.
 *
 */
class QualifiedId {
  /**
   * Create a QualifiedId.
   * @member {string} [id]
   * @member {string} [scope]
   * @member {string} [type] Possible values include: 'Group', 'Portfolio',
   * 'ReferencePortfolio'
   */
  constructor() {
  }

  /**
   * Defines the metadata of QualifiedId
   *
   * @returns {object} metadata of QualifiedId
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'QualifiedId',
      type: {
        name: 'Composite',
        className: 'QualifiedId',
        modelProperties: {
          id: {
            required: false,
            readOnly: true,
            serializedName: 'id',
            type: {
              name: 'String'
            }
          },
          scope: {
            required: false,
            readOnly: true,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          },
          type: {
            required: false,
            readOnly: true,
            serializedName: 'type',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = QualifiedId;
