// lusid-client/models/referencePortfolioConstituent.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a ReferencePortfolioConstituent.
 */
class ReferencePortfolioConstituent {
  /**
   * Create a ReferencePortfolioConstituent.
   * @member {object} [id] Identifier for this constituent
   * @member {string} [id.uid]
   * @member {array} [properties]
   * @member {number} [quantity]
   * @member {string} [type] Possible values include: 'Shares', 'Weight',
   * 'Nominal'
   */
  constructor() {
  }

  /**
   * Defines the metadata of ReferencePortfolioConstituent
   *
   * @returns {object} metadata of ReferencePortfolioConstituent
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ReferencePortfolioConstituent',
      type: {
        name: 'Composite',
        className: 'ReferencePortfolioConstituent',
        modelProperties: {
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'Composite',
              className: 'SecurityUid'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          },
          quantity: {
            required: false,
            serializedName: 'quantity',
            type: {
              name: 'Number'
            }
          },
          type: {
            required: false,
            serializedName: 'type',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ReferencePortfolioConstituent;
