// lusid-client/models/referencePortfolioResponse.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a ReferencePortfolioResponse.
 */
class ReferencePortfolioResponse {
  /**
   * Create a ReferencePortfolioResponse.
   * @member {string} [self]
   * @member {string} [schema]
   * @member {object} [error]
   * @member {number} [error.status] The status code that will be returned to
   * the client
   * @member {string} [error.code] The Finbourne specific error-code that
   * encapsulates the specific issue encountered. Possible values include:
   * 'PersonalisationNotFound', 'NonRecursivePersonalisation',
   * 'VersionNotFound', 'TradeNotFound', 'SecurityNotFound',
   * 'PropertyNotFound', 'PortfolioRecursionDepth',
   * 'PortfolioPropertiesNotFound', 'GroupNotFound', 'PortfolioNotFound',
   * 'EntitySchemaNotFound', 'PortfolioNameNotFound', 'PortfolioAlreadyExists',
   * 'OrphanedPortfolio', 'MultiplePropertyDefinitionNotFound',
   * 'InvalidProperty', 'InvalidBasePortfolioAction', 'InsufficientAuthClaims',
   * 'MissingBaseClaims', 'MissingIdentity', 'PropertyNotDefined',
   * 'CannotDeleteSystemProperty', 'CannotModifyImmutablePropertyField',
   * 'PropertyAlreadyExists', 'InvalidPropertyLifeTime',
   * 'RequiredFieldIsMissing', 'CannotModifyDefaultPropertyFormat',
   * 'GroupAlreadyExists', 'NoSuchPropertyDataFormat',
   * 'CannotAddMissingItemToGroup', 'CannotAddAGroupToItself',
   * 'ValidationError', 'LoopDetectedInGroupHierarchy',
   * 'AddingSubGroupCreatedALoop', 'SubGroupAlreadyExists',
   * 'PortfolioAlreadyExistsInGroup', 'MissingClassificationData',
   * 'PriceSourceNotFound', 'AnalyticStoreNotFound', 'InvalidDateFormat',
   * 'AnalyticStoreAlreadyExists', 'ClientSecurityAlreadyExists',
   * 'DuplicateParameterException', 'OneOrMoreFailures', 'OneOrMoreMissing',
   * 'ResultsNotFound', 'OrderFieldNotInResultSet', 'OperationFailedEnum',
   * 'ElasticSearchError', 'InvalidParameterValue', 'Unknown'
   * @member {string} [error.property] The part of the resource which provoked
   * the error (e.g. field)
   * @member {string} [error.message] The non-technical-user friendly message
   * describing the error and how it might be remedied.
   * @member {string} [error.developerMessage] A technical error message that
   * contains the details of the issue and how it might be fixed.
   * @member {string} [error.moreInfo] A link to the endpoint that can provide
   * the dev with more information about that class of error.
   * @member {object} [meta]
   * @member {number} [meta.duration]
   * @member {boolean} [meta.success]
   * @member {string} [meta.requestId]
   * @member {object} [referencePortfolio]
   * @member {string} [referencePortfolio.id] Portfolio id, must be unique
   * across all portfolios
   * @member {string} [referencePortfolio.name] Portfolio name, must be unique
   * across all portfolios
   * @member {date} [referencePortfolio.created] The original creation date,
   * not the effective date
   * of subsequent modification.
   * @member {string} [referencePortfolio.referencePortfolioId] If creating a
   * derived portfolio, this is the portfolio id of the reference portfolio
   * @member {string} [referencePortfolio.referenceScope] If creating a derived
   * portfolio, this is the scope of the reference portfolio
   * @member {string} [referencePortfolio.type] Possible values include:
   * 'Portfolio', 'ReferencePortfolio'
   */
  constructor() {
  }

  /**
   * Defines the metadata of ReferencePortfolioResponse
   *
   * @returns {object} metadata of ReferencePortfolioResponse
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ReferencePortfolioResponse',
      type: {
        name: 'Composite',
        className: 'ReferencePortfolioResponse',
        modelProperties: {
          self: {
            required: false,
            serializedName: 'self',
            type: {
              name: 'String'
            }
          },
          schema: {
            required: false,
            serializedName: 'schema',
            type: {
              name: 'String'
            }
          },
          error: {
            required: false,
            serializedName: 'error',
            type: {
              name: 'Composite',
              className: 'ErrorMessage'
            }
          },
          meta: {
            required: false,
            serializedName: 'meta',
            type: {
              name: 'Composite',
              className: 'ResponseMeta'
            }
          },
          referencePortfolio: {
            required: false,
            serializedName: 'referencePortfolio',
            type: {
              name: 'Composite',
              className: 'Portfolio'
            }
          }
        }
      }
    };
  }
}

module.exports = ReferencePortfolioResponse;
