// lusid-client/models/resourceListPortfolioDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a ResourceListPortfolioDto.
 */
class ResourceListPortfolioDto {
  /**
   * Create a ResourceListPortfolioDto.
   * @member {array} [values]
   * @member {string} [href]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ResourceListPortfolioDto
   *
   * @returns {object} metadata of ResourceListPortfolioDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ResourceList_PortfolioDto_',
      type: {
        name: 'Composite',
        className: 'ResourceListPortfolioDto',
        modelProperties: {
          values: {
            required: false,
            serializedName: 'values',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PortfolioDtoElementType',
                  type: {
                    name: 'Composite',
                    className: 'PortfolioDto'
                  }
              }
            }
          },
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = ResourceListPortfolioDto;
