// lusid-client/models/responsePagination.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a ResponsePagination.
 */
class ResponsePagination {
  /**
   * Create a ResponsePagination.
   * @member {boolean} [finalPage]
   * @member {number} [count]
   * @member {number} [totalItems]
   * @member {number} [page]
   * @member {string} [next]
   * @member {string} [previous]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ResponsePagination
   *
   * @returns {object} metadata of ResponsePagination
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ResponsePagination',
      type: {
        name: 'Composite',
        className: 'ResponsePagination',
        modelProperties: {
          finalPage: {
            required: false,
            serializedName: 'finalPage',
            type: {
              name: 'Boolean'
            }
          },
          count: {
            required: false,
            serializedName: 'count',
            type: {
              name: 'Number'
            }
          },
          totalItems: {
            required: false,
            serializedName: 'totalItems',
            type: {
              name: 'Number'
            }
          },
          page: {
            required: false,
            serializedName: 'page',
            type: {
              name: 'Number'
            }
          },
          next: {
            required: false,
            serializedName: 'next',
            type: {
              name: 'String'
            }
          },
          previous: {
            required: false,
            serializedName: 'previous',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ResponsePagination;
