// lusid-client/models/responseMeta.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a ResponseMeta.
 */
class ResponseMeta {
  /**
   * Create a ResponseMeta.
   * @member {number} [duration]
   * @member {boolean} [success]
   * @member {string} [requestId]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ResponseMeta
   *
   * @returns {object} metadata of ResponseMeta
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ResponseMeta',
      type: {
        name: 'Composite',
        className: 'ResponseMeta',
        modelProperties: {
          duration: {
            required: false,
            serializedName: 'duration',
            type: {
              name: 'Number'
            }
          },
          success: {
            required: false,
            serializedName: 'success',
            type: {
              name: 'Boolean'
            }
          },
          requestId: {
            required: false,
            serializedName: 'requestId',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ResponseMeta;
