// lusid-client/models/resultDataSchema.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a ResultDataSchema.
 */
class ResultDataSchema {
  /**
   * Create a ResultDataSchema.
   * @member {array} [nodeValueSchema]
   * @member {object} [propertySchema]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ResultDataSchema
   *
   * @returns {object} metadata of ResultDataSchema
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ResultDataSchema',
      type: {
        name: 'Composite',
        className: 'ResultDataSchema',
        modelProperties: {
          nodeValueSchema: {
            required: false,
            serializedName: 'nodeValueSchema',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'KeyValuePairPropertyKeyFieldSchemaElementType',
                  type: {
                    name: 'Composite',
                    className: 'KeyValuePairPropertyKeyFieldSchema'
                  }
              }
            }
          },
          propertySchema: {
            required: false,
            serializedName: 'propertySchema',
            type: {
              name: 'Dictionary',
              value: {
                  required: false,
                  serializedName: 'FieldSchemaElementType',
                  type: {
                    name: 'Composite',
                    className: 'FieldSchema'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = ResultDataSchema;
