// lusid-client/models/schemaDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a SchemaDto.
 */
class SchemaDto {
  /**
   * Create a SchemaDto.
   * @member {string} [href]
   * @member {array} [items]
   */
  constructor() {
  }

  /**
   * Defines the metadata of SchemaDto
   *
   * @returns {object} metadata of SchemaDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'SchemaDto',
      type: {
        name: 'Composite',
        className: 'SchemaDto',
        modelProperties: {
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          items: {
            required: false,
            serializedName: 'items',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'KeyValuePairStringFieldSchemaElementType',
                  type: {
                    name: 'Composite',
                    className: 'KeyValuePairStringFieldSchema'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = SchemaDto;
