// lusid-client/models/scopedIdentifier.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class for an external identifier, ie one containing a scope and a name
 * See https://github.com/finbourne/wiki/wiki/Identifiers
 *
 */
class ScopedIdentifier {
  /**
   * Create a ScopedIdentifier.
   * @member {string} [scope]
   * @member {string} [name]
   */
  constructor() {
  }

  /**
   * Defines the metadata of ScopedIdentifier
   *
   * @returns {object} metadata of ScopedIdentifier
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'ScopedIdentifier',
      type: {
        name: 'Composite',
        className: 'ScopedIdentifier',
        modelProperties: {
          scope: {
            required: false,
            readOnly: true,
            serializedName: 'scope',
            type: {
              name: 'String'
            }
          },
          name: {
            required: false,
            readOnly: true,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = ScopedIdentifier;
