// lusid-client/models/securityAnalyticData.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a SecurityAnalyticData.
 */
class SecurityAnalyticData {
  /**
   * Create a SecurityAnalyticData.
   * @member {object} [id]
   * @member {string} [id.uid]
   * @member {number} [value]
   */
  constructor() {
  }

  /**
   * Defines the metadata of SecurityAnalyticData
   *
   * @returns {object} metadata of SecurityAnalyticData
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'SecurityAnalyticData',
      type: {
        name: 'Composite',
        className: 'SecurityAnalyticData',
        modelProperties: {
          id: {
            required: false,
            serializedName: 'id',
            type: {
              name: 'Composite',
              className: 'SecurityUid'
            }
          },
          value: {
            required: false,
            serializedName: 'value',
            type: {
              name: 'Number'
            }
          }
        }
      }
    };
  }
}

module.exports = SecurityAnalyticData;
