// lusid-client/models/securityClassificationData.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a SecurityClassificationData.
 */
class SecurityClassificationData {
  /**
   * Create a SecurityClassificationData.
   * @member {object} [uid]
   * @member {string} [uid.uid]
   * @member {date} [effectiveDate]
   * @member {array} [properties]
   */
  constructor() {
  }

  /**
   * Defines the metadata of SecurityClassificationData
   *
   * @returns {object} metadata of SecurityClassificationData
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'SecurityClassificationData',
      type: {
        name: 'Composite',
        className: 'SecurityClassificationData',
        modelProperties: {
          uid: {
            required: false,
            serializedName: 'uid',
            type: {
              name: 'Composite',
              className: 'SecurityUid'
            }
          },
          effectiveDate: {
            required: false,
            serializedName: 'effectiveDate',
            type: {
              name: 'DateTime'
            }
          },
          properties: {
            required: false,
            serializedName: 'properties',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'PropertyElementType',
                  type: {
                    name: 'Composite',
                    className: 'Property'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = SecurityClassificationData;
