// lusid-client/models/tradesDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a TradesDto.
 */
class TradesDto {
  /**
   * Create a TradesDto.
   * @member {array} [values]
   * @member {string} [href]
   * @member {object} [version]
   * @member {date} [version.validFromDate] The time at which this version
   * became valid.
   * @member {object} [version.asAtDate] The time at which this version was
   * durably persisted.
   * @member {string} [version.updatedBy] The name of the person or system
   * which created this version.
   * @member {string} [version.href]
   * @member {array} [version._links]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of TradesDto
   *
   * @returns {object} metadata of TradesDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'TradesDto',
      type: {
        name: 'Composite',
        className: 'TradesDto',
        modelProperties: {
          values: {
            required: false,
            serializedName: 'values',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'TradeDtoElementType',
                  type: {
                    name: 'Composite',
                    className: 'TradeDto'
                  }
              }
            }
          },
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          version: {
            required: false,
            serializedName: 'version',
            type: {
              name: 'Composite',
              className: 'VersionDto'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = TradesDto;
