// lusid-client/models/updatePortfolioRequest.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Class representing a UpdatePortfolioRequest.
 */
class UpdatePortfolioRequest {
  /**
   * Create a UpdatePortfolioRequest.
   * @member {string} name Portfolio name (aka description
   * @member {date} [created] The original creation date, not the effective
   * date
   * of subsequent modification.
   */
  constructor() {
  }

  /**
   * Defines the metadata of UpdatePortfolioRequest
   *
   * @returns {object} metadata of UpdatePortfolioRequest
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'UpdatePortfolioRequest',
      type: {
        name: 'Composite',
        className: 'UpdatePortfolioRequest',
        modelProperties: {
          name: {
            required: true,
            serializedName: 'name',
            type: {
              name: 'String'
            }
          },
          created: {
            required: false,
            serializedName: 'created',
            type: {
              name: 'DateTime'
            }
          }
        }
      }
    };
  }
}

module.exports = UpdatePortfolioRequest;
