// lusid-client/models/upsertPortfolioTradesDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Class representing a UpsertPortfolioTradesDto.
 */
class UpsertPortfolioTradesDto {
  /**
   * Create a UpsertPortfolioTradesDto.
   * @member {string} [href]
   * @member {boolean} [validationFailed]
   * @member {array} [validationErrors]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of UpsertPortfolioTradesDto
   *
   * @returns {object} metadata of UpsertPortfolioTradesDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'UpsertPortfolioTradesDto',
      type: {
        name: 'Composite',
        className: 'UpsertPortfolioTradesDto',
        modelProperties: {
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          validationFailed: {
            required: false,
            serializedName: 'validationFailed',
            type: {
              name: 'Boolean'
            }
          },
          validationErrors: {
            required: false,
            serializedName: 'validationErrors',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'ObjectElementType',
                  type: {
                    name: 'Object'
                  }
              }
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = UpsertPortfolioTradesDto;
