// lusid-client/models/versionDto.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

const models = require('./index');

/**
 * Describes the version metadata of an entity.
 *
 */
class VersionDto {
  /**
   * Create a VersionDto.
   * @member {date} [validFromDate] The time at which this version became
   * valid.
   * @member {object} [asAtDate] The time at which this version was durably
   * persisted.
   * @member {string} [updatedBy] The name of the person or system which
   * created this version.
   * @member {string} [href]
   * @member {array} [_links]
   */
  constructor() {
  }

  /**
   * Defines the metadata of VersionDto
   *
   * @returns {object} metadata of VersionDto
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'VersionDto',
      type: {
        name: 'Composite',
        className: 'VersionDto',
        modelProperties: {
          validFromDate: {
            required: false,
            serializedName: 'validFromDate',
            type: {
              name: 'DateTime'
            }
          },
          asAtDate: {
            required: false,
            serializedName: 'asAtDate',
            type: {
              name: 'Object'
            }
          },
          updatedBy: {
            required: false,
            serializedName: 'updatedBy',
            type: {
              name: 'String'
            }
          },
          href: {
            required: false,
            serializedName: 'href',
            type: {
              name: 'String'
            }
          },
          _links: {
            required: false,
            serializedName: '_links',
            type: {
              name: 'Sequence',
              element: {
                  required: false,
                  serializedName: 'LinkElementType',
                  type: {
                    name: 'Composite',
                    className: 'Link'
                  }
              }
            }
          }
        }
      }
    };
  }
}

module.exports = VersionDto;
