// lusid-client/models/version.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * Describes the version metadata of an entity.
 *
 */
class Version {
  /**
   * Create a Version.
   * @member {number} [number] The monotonically-increasing version number for
   * this individual version.
   * @member {date} [validFromDate] The time at which this version became
   * valid.
   * @member {date} [validToDate] The time at which this version ceased to be
   * valid.
   * If this is most recent version, this value will be set to
   * DateTimeOffset.MaxValue.
   * @member {object} [asAtDate] The time at which this version was durably
   * persisted.
   * @member {string} [updatedBy] The name of the person or system which
   * created this version.
   */
  constructor() {
  }

  /**
   * Defines the metadata of Version
   *
   * @returns {object} metadata of Version
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'Version',
      type: {
        name: 'Composite',
        className: 'Version',
        modelProperties: {
          number: {
            required: false,
            readOnly: true,
            serializedName: 'number',
            type: {
              name: 'Number'
            }
          },
          validFromDate: {
            required: false,
            readOnly: true,
            serializedName: 'validFromDate',
            type: {
              name: 'DateTime'
            }
          },
          validToDate: {
            required: false,
            readOnly: true,
            serializedName: 'validToDate',
            type: {
              name: 'DateTime'
            }
          },
          asAtDate: {
            required: false,
            readOnly: true,
            serializedName: 'asAtDate',
            type: {
              name: 'Object'
            }
          },
          updatedBy: {
            required: false,
            readOnly: true,
            serializedName: 'updatedBy',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = Version;
