// lusid-client/models/webLogMessage.js
/*
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */

'use strict';

/**
 * A log message structured for provision by a web project
 *
 */
class WebLogMessage {
  /**
   * Create a WebLogMessage.
   * @member {string} [version] The semantic version of the remote application
   * submitting the log
   * @member {string} [url] The url of the resource from which the message
   * originated
   * @member {string} [message] The body of the message
   * @member {string} [context] Context as to the occurance of the message
   * @member {string} [severity] The severity of the message. Possible values
   * include: 'Warn', 'Error'
   * @member {string} [stacktrace] Any stacktrace that may be relavent
   * @member {string} [browser] Any browser/user-agent/os related context
   */
  constructor() {
  }

  /**
   * Defines the metadata of WebLogMessage
   *
   * @returns {object} metadata of WebLogMessage
   *
   */
  mapper() {
    return {
      required: false,
      serializedName: 'WebLogMessage',
      type: {
        name: 'Composite',
        className: 'WebLogMessage',
        modelProperties: {
          version: {
            required: false,
            readOnly: true,
            serializedName: 'version',
            type: {
              name: 'String'
            }
          },
          url: {
            required: false,
            readOnly: true,
            serializedName: 'url',
            type: {
              name: 'String'
            }
          },
          message: {
            required: false,
            readOnly: true,
            serializedName: 'message',
            type: {
              name: 'String'
            }
          },
          context: {
            required: false,
            readOnly: true,
            serializedName: 'context',
            type: {
              name: 'String'
            }
          },
          severity: {
            required: false,
            readOnly: true,
            serializedName: 'severity',
            type: {
              name: 'String'
            }
          },
          stacktrace: {
            required: false,
            readOnly: true,
            serializedName: 'stacktrace',
            type: {
              name: 'String'
            }
          },
          browser: {
            required: false,
            readOnly: true,
            serializedName: 'browser',
            type: {
              name: 'String'
            }
          }
        }
      }
    };
  }
}

module.exports = WebLogMessage;
